massa-deferred-calls/src/slot_changes.rs

```diff
@@ -0,0 +1,296 @@
+use std::collections::BTreeMap;
+
+use crate::{
+    call::{DeferredCallDeserializer, DeferredCallSerializer},
+    DeferredCall, DeferredRegistryBaseFeeChange, DeferredRegistryCallChange,
+    DeferredRegistryGasChange,
+};
+use massa_ledger_exports::{
+    SetOrDelete, SetOrDeleteDeserializer, SetOrDeleteSerializer, SetOrKeepDeserializer,
+    SetOrKeepSerializer,
+};
+use massa_models::{
+    amount::{Amount, AmountDeserializer, AmountSerializer},
+    deferred_call_id::{DeferredCallId, DeferredCallIdDeserializer, DeferredCallIdSerializer},
+};
+use massa_serialization::{
+    Deserializer, SerializeError, Serializer, U64VarIntDeserializer, U64VarIntSerializer,
+};
+use nom::{
+    error::{context, ContextError, ParseError},
+    multi::length_count,
+    sequence::tuple,
+    IResult, Parser,
+};
+use serde::{Deserialize, Serialize};
+use std::ops::Bound::Included;
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct DeferredRegistrySlotChanges {
+    pub calls: BTreeMap<DeferredCallId, DeferredRegistryCallChange>,
+    pub gas: DeferredRegistryGasChange<u64>,
+    pub base_fee: DeferredRegistryBaseFeeChange,
+}
+
+impl DeferredRegistrySlotChanges {
+    pub fn calls_len(&self) -> usize {
+        self.calls.len()
+    }
+
+    pub fn merge(&mut self, _other: DeferredRegistrySlotChanges) {
+        unimplemented!("DeferredRegistrySlotChanges::merge")
```

Comment:
will do later?

Reply:
i haven't found any use case for this function yet